plone/app/contenttypes/migration/utils.py

```diff
@@ -207,6 +207,12 @@ def migrate_leadimage(source_object, target_object):
                     "Could not migrate collective.leadimage fields.")
         return
 
+    acc = source_object.getField(
+        OLD_LEADIMAGE_FIELD_NAME).getAccessor(source_object)()
+    if not acc.filename:
+        # skip if old content has field but has no lead image in the field
```

Comment:
I'm not sure but is acc is ever None but it would fail with an AttributeError. Better use
``if getattr(acc, 'filename', None) is not None:``